website/server/models/group.js

```diff
@@ -833,21 +833,24 @@ schema.statics.tavernBoss = async function tavernBoss (user, progress) {
   }
 };
 
-schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
+schema.methods.leave = async function leaveGroup (user, keep = 'keep-all', keepChallenges = 'leave-challenges') {
   let group = this;
   let update = {
     $inc: {memberCount: -1},
   };
 
-  let challenges = await Challenge.find({
-    _id: {$in: user.challenges},
-    group: group._id,
-  });
+  // only remove user from challenges if it's set to leave-challenges
+  if (keepChallenges === 'leave-challenges') {
+    let challenges = await Challenge.find({
+      _id: {$in: user.challenges},
+      group: group._id,
+    });
```

Comment:
missing .exec() (Note: I'm commenting on all PRs that have missing `.exec()` calls, not reviewing entire PRs)